### ISSUE TITLE
match_pft fails when pft is not in the xml

### ISSUE THREAD

**Issue by istfer** [PecanProject/pecan#2372]
## Description

**Is your feature request related to a problem? Please describe.**
People who worked with [match_pft](https://github.com/PecanProject/pecan/blob/develop/modules/data.land/R/match_pft.R) function might know that it can fail in various ways. Sometimes the species is not matched to any PFTs in the database, or one species might be associated with multiple PFTs that are requested for that run. These require fixes in the DB and there isn't much the function itself can do.

But it can also fail when the PFT that the species is associated with is not listed under the pft tag of your xml. E.g  if there is a Tsuga Canadensis in the plot, but if you didn't include "temperate.Late_Conifer" in your xml for ED2 runs, the code can't match it and will fail. So you need to go back, add temperate.Late_Conifer under your pft tag and rerun. Which is fine, but the function only says "Unmatched species Tsuga Canadensis". This example is trivial, but sometimes there can be tens of unmatched species and you might not be familiar with all of them. Then, you need to check which PFTs they're matched with in the DB and add those to your xml and rerun which can get really time consuming. 

This is especially an issue when querying FIA DB for generating IC files, because you cannot exactly know which species (and therefore PFTs) are gonna be there before starting the IC processing.

## Proposed Solution

**Describe the solution you'd like**
I was wondering if instead of failing, can the function automatically populate the settings pft list, or at least give a lot more informative error message like "X, Y, Z spp unmatched. Add K, L, M PFTs to your xml and rerun" This could possibly save a lot of time for people.

## Alternatives Considered

**Describe alternatives you've considered**
Alternatively, those xmls for ED runs (or any model with multiple PFTs) can include all pfts in the beginning and the function can filter down the PFTs that don't exist in that site.

@ashiklom @mdietze what would you think?

**Comment by ashiklom** [comment]
> give a lot more informative error message like "X, Y, Z spp unmatched. Add K, L, M PFTs to your xml and rerun"

This is a great idea, and a relatively easy fix. I guess the `K, L, M` PFTs would just be all the PFTs that have the missing species? Perhaps that can be optionally subset by `modeltype`, which can be made an optional argument to `match_pft`? E.g...

```r
# pseudocode -- won't actually work
match_pft <- function(..., modeltype = NULL) {
  # ...
  if (<species unmatched>) {
    pft_query <- tbl(con, "species") %>%
      filter(<unmatched species>) %>%
      inner_join(tbl(con, "pfts_species")) %>%
      inner_join(tbl(con, "pfts"))
    if (!is.null(modeltype)) pft_query <- filter(pft_query, model_type == !!modeltype)
    pft_table <- pft_query %>% select(scientificname, pft, species_id, pft_id) %>% collect()
    logger.severe("<unamtched species> species not matched to input PFTs. ",
                  "See the following table for other PFTs you might consider adding:\n",
                  print2string(pft_table, n = Inf))
  }
}
``` 

> can the function automatically populate the settings pft list

I'm more hesitant about this for a few reasons:

- (1) It makes `match_pfts` do a lot more than its name suggests, and I'm against functions that do too many things as a general rule. If we did want to have automated PFT configuration like this, I would rather it be a wrapper called something like `configure_settings_pfts` that calls `match_pfts` as part of its execution.
- (2) If there are multiple PFTs containing the missing species, how would you know exactly which one to add? We have a _lot_ of different PFTs with highly overlapping (if not identical) species mappings, so I don't think there's a good mechanism for picking the "best" one. One thing we _could_ consider (down the road; perhaps for PEcAn 2.0) is creating a "PFT groups" table/column, at which point functionality like this would be much less ambiguous.

**Comment by istfer** [comment]
>It makes match_pfts do a lot more than its name suggests

Agreed. Modifying settings could also result in unexpected things (for the user) downstream.

Bu that informative error message would be super helpful!
